tasks.c

```diff
@@ -956,7 +958,9 @@ static void prvYieldForTask( TCB_t * pxTCB,
             }
 
             configASSERT( ( uxCurrentPriority > tskIDLE_PRIORITY ) || ( xTaskScheduled == pdTRUE ) );
-            uxCurrentPriority--;
+            if(uxCurrentPriority > 0) {
+                uxCurrentPriority--;
+            }
```

Comment:
Similarly here as well

```suggestion
            /* The top ready priority cannot be less than the idle task's priority. */
            if( uxCurrentPriority > tskIDLE_PRIORITY )
            {
                uxCurrentPriority--;
            }
```